unblob/handlers/filesystem/netgearchk.py

```diff
@@ -0,0 +1,76 @@
+import io
+from pathlib import Path
+from typing import Optional
+
+from structlog import get_logger
+
+from unblob.extractor import carve_chunk_to_file
+from unblob.file_utils import Endian, File, StructParser
+from unblob.models import Chunk, Extractor, HexString, StructHandler, ValidChunk
+
+logger = get_logger()
+
+CHK_HEADER = r"""
+        typedef struct chk_header {
+            uint32 magic;           /* "2A 23 24 5E" */
+            uint32 header_len;
+            uint8  reserved[8];
+            uint32 kernel_chksum;
+            uint32 rootfs_chksum;
+            uint32 kernel_len;
+            uint32 rootfs_len;
+            uint32 image_chksum;
+            uint32 header_chksum;
+            /* char board_id[total_header_len - header_len] upto MAX_BOARD_ID_LEN */
+            /* uint32 total_header_len; */
```

Comment:
Turns out the `header_len` is the header size *including the variable length board_id*. So you can do something like this:

 ```suggestion
            char board_id[header_len-40];
```

You can even be more explicit with something along those lines (can be defined above the struct):

```c
#define FIXED_HEADER_LEN 40
```

And use it this way:

 ```c
char board_id[header_len-FIXED_HEADER_LEN];
```

Reply:
We set a fix header len of 40, this way we could uncomment the board_id.
char board_id[header_len - FIXED_HEADER_LEN]; /* upto MAX_BOARD_ID_LEN */